command/agent/operator_endpoint.go

```diff
@@ -49,21 +55,222 @@ func (s *HTTPServer) OperatorRaftPeer(resp http.ResponseWriter, req *http.Reques
 		return nil, nil
 	}
 
-	var args structs.RaftPeerByAddressRequest
-	s.parseWriteRequest(req, &args.WriteRequest)
-
 	params := req.URL.Query()
-	if _, ok := params["address"]; ok {
-		args.Address = raft.ServerAddress(params.Get("address"))
-	} else {
+	_, hasID := params["id"]
+	_, hasAddress := params["address"]
+
+	if !hasID && !hasAddress {
+		resp.WriteHeader(http.StatusBadRequest)
+		fmt.Fprint(resp, "Must specify either ?id with the server's ID or ?address with IP:port of peer to remove")
+		return nil, nil
```

Comment:
Change to be `return nil, CodedError(http.StatusBadRequest, error_string)` here and else where were this is used